javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -36,6 +40,14 @@
 @SuppressWarnings({"unused", "SystemOut"})
 public class ExtensionClassLoader extends URLClassLoader {
   public static final String EXTENSIONS_CONFIG = "otel.javaagent.extensions";
+  // if this class was defined with all permissions then also define classes in this class loader
+  // with all permissions
+  // this class is defined with all permissions when security manager support is enabled
+  private static final boolean addAllPermissions =
+      ExtensionClassLoader.class
+          .getProtectionDomain()
+          .getPermissions()
+          .implies(new AllPermission());
```

Comment:
why not pass in `isSecurityManagerSupportEnabled` instead?

Reply:
I initially wanted to avoid passing `isSecurityManagerSupportEnabled` through a lot of methods, but it really isn't too bad, changed it.